### ISSUE TITLE
Maybe a simple update? 

### ISSUE THREAD

**Issue by K4viar** [UnknownCollections/ExponentialIdle-SaveGameEditor#3]
Unfortunately this program seems to not be working properly in the current version of the game (138), when trying to save what I edited this error message show up: "One or more inputs were not valid and have marked". I don't have the expertise to troubleshoot this myself, so I'm posting here in hopes that someone can bring back to life that amazing project of this even greater game!

**Comment by rorik** [comment]
Not much we can do without having the source for `crypto.obfuscate.js`.
I'd love to give it a spin a figure out the difference in save data between versions 123 and 138, but I'm not going to spend countless hours trying to reconstruct the Webpacked/minified/obfuscated code, nor going to bother trying to find the encryption keys/steps by reverse engineering the game code, nor going to crack the password for the `crypto.obfuscate.js.rar` file.

While I somewhat understand the point of having the keys hidden, this makes the project unusable for people that aren't already familiar with how the saves are structured and encrypted (I assume this is everyone expect @conicgames and @UnknownCollections).
Obfuscated OSS projects will eventually die once the only people (or just 1 in this case) who have access to the full source code move to a different thing. Seeing as the last commit was 11 months ago, I really hope this is not already the case.

As someone who doesn't use the leaderboards, having this level of protection for an offline game seems overkill.
This is more of a suggestion/rant to the game dev(s), but please consider making the game more cheat friendly, make it an irreversible toggle at the app/device level, such that I can have unencrypted saves but with online features disabled. It can even be behind a (reasonable) paywall.
Considering you can time skip by changing the device time, I assume this kind of cheat flagging already exists (I don't know if it disables the leaderboards since I'm not signed in to begin with).
People want to cheat, people want to play at their own pace. Adding simple anti-cheat measures is fine, it will deter 99% of casual players, but overly engineering the save system will just make some people stop playing altogether since they don't want to have to wait for days/weeks to have any meaningful progress.

Sorry for the long rant, I'm not blaming anyone nor saying that the system has to be changed. Ultimately the end decision is up to the game dev and it should be respected. Thank you both for taking time to make the game and tool.

**Comment by K4viar** [comment]
Yes, I completely agree with your statements. At the end all we can do is hope that the game developers do something about this kinda nonsense anti-cheat behavior in an offline game that have zero to none competitiveness, or that the tool dev releases an update and/or the source code of  `crypto.obfuscate.js`

In my specific case, I just wanted to accelerate the pace of the game cause sometimes I found it veeeeryy slow to progress. 

(By the way, the time skip is indeed flagged, but you have a "chance of redemption" and can restart the game from scratch, this also does not affect the leaderboards)

**Comment by EzequielDM** [comment]
At this point it seems easier to reverse engineer the game rather than edit the save games. Obfuscated Open Source is a joke at the least.

**Comment by conicgames** [comment]
Hello,

I'm the developer of Exponential Idle. I'm not affiliated to this save editor but the author contacted me before releasing it to the public and I'm the one asking for the encryption key to be obfuscated. If you want to blame someone, blame me.

Please stay respectful in discussions, I feel frustration starting to grow from message to message toward someone who offered this tool for free and that spent a lot of time to do it.

I did not share at any point the encryption key with the author of this save editor. He indeed reverse engineered the app and got it by himself, and figured out everything that the save contains. Many people did it.

That being said, I agree that an update to this tool is necessary. I'll try to contact the author and see if he has time to add the missing fields that were added in the latest releases.

If not, I'll try to update it myself and make a pull request.

Thanks for your understanding.

**Comment by K4viar** [comment]
Thanks for the response and sorry if any of my statements sounds disrespectful, it sure was not the intention. I still disagree with the decision to make the encryption obfuscated, but I can 100% understand the reasons of why you would want that.

Since my first post here I was able to finish the game and indeed is amazing, I absolutely crave for more content in the future!

That being said, I would still like an update to this tool, the students and star optimizer integration sure is handy, and some extra speed would be awesome to pass through some custom theories milestones.

Thanks for your time and I hope the tool dev have time to work here again in the following weeks/months😉

**Comment by conicgames** [comment]
I tried to contact the author but I got no response (via Discord & GitHub). I think I'll have to update it myself but I'll first have to recreate the `crypto.obfuscate.js` file first. The major inconvenient of this is that I won't be able to create a pull request to update the website if I cannot communicate with the author so I might have to host it on the main game website. This might create confusion in the future until the main game website gets referenced first.

As the author said, I could consider sharing crypto.obfuscate.js on request. But I won't make it publicly available.
